commodore/config.py

```diff
@@ -104,7 +107,19 @@ def refs_dir(self) -> P:
     @property
     def api_token(self):
         if self._api_token is None and self.api_url:
-            self._api_token = tokencache.get(self.api_url)
+            token = tokencache.get(self.api_url)
+            if token is not None:
+                # We don't verify the signature, we just want to know if the token is expired
+                # lieutenant will decide if it's valid
+                try:
+                    t = jwt.decode(
+                        token, algorithms=["RS256"], options={"verify_signature": False}
+                    )
+                    if "exp" in t and t["exp"] < time.time():
```

Comment:
This comparison leaves a slight chance of ending up trying to use an expired token if the token expires in a few seconds, and we perform the actual API calls after it's expired.

Maybe better:

```suggestion
                    if "exp" in t and t["exp"] < time.time() + 10:
```